test/support/sycl_alloc_utils.h

```diff
@@ -126,40 +150,109 @@ class usm_data_transfer
         return __ptr;
     }
 
-    template<typename _Iterator>
+    template <typename _Iterator>
+    void update_data(_Iterator __it)
+    {
+        update_data(__it, 0, __count);
+    }
+
+    template<typename _Iterator, typename _Size>
+    void update_data(_Iterator __it, __difference_type __offset, _Size __objects_count)
+    {
+        assert(0 <= __offset);
+        assert(0 <= __objects_count);
+        assert(__offset + __objects_count <= __count);
+
+        if (__count > 0 && __objects_count > 0)
+        {
+            assert(__ptr != nullptr);
+
+            //TODO: support copying data provided by non-contiguous iterator
+            auto __src = std::addressof(*__it);
+            assert(std::addressof(*(__it + __objects_count)) - __src == __objects_count);
+
+            copy_data_impl(__src, __ptr + __offset, __objects_count);
+        }
+    }
+
+    template <typename _Iterator>
     void retrieve_data(_Iterator __it)
     {
-        if (__count > 0)
+        return retrieve_data(__it, 0, __count);
+    }
+
+    template<typename _Iterator>
+    void retrieve_data(_Iterator __it, __difference_type __offset, __difference_type __objects_count)
```

Comment:
Why do we need such version of retrieve_data?
The all usages of retrieve_data accept just one argument (Iterator type). At least, I cannot see the other entries.

Reply:
This version used in class RetrieveUsmData:
obj.retrieve_data(it_from, offset, count);